internal/model/historical_recommendation_set.go

```diff
@@ -22,12 +24,16 @@ type HistoricalRecommendationSet struct {
 func (r *HistoricalRecommendationSet) CreateHistoricalRecommendationSet() error {
 	db := database.GetDB()
 	result := db.Clauses(clause.OnConflict{
-		Columns:   []clause.Column{{Name: "workload_id"}, {Name: "container_name"}, {Name: "monitoring_end_time"}},
+		Columns:   []clause.Column{{Name: "org_id"}, {Name: "workload_id"}, {Name: "container_name"}, {Name: "monitoring_end_time"}},
 		DoUpdates: clause.AssignmentColumns([]string{"monitoring_start_time", "monitoring_end_time", "recommendations", "updated_at"}),
 	}).Create(r)
 
 	if result.Error != nil {
-		dbError.Inc()
+		if strings.Contains(result.Error.Error(), "no partition") {
+			partitionMissing.Inc()
+		} else {
+			dbError.Inc()
+		}
```

Comment:
```suggestion
		}
		dbError.Inc()		
```

I would prefer to keep this outside of else block as it is also a DB error.